go/service/user.go

```diff
@@ -344,6 +345,248 @@ func (h *UserHandler) UploadUserAvatar(ctx context.Context, arg keybase1.UploadU
+				PickerSubtext: serviceType.GetTypeName(), // xxx url or mastodon
+				PickerIcon:    dummyIcon,
+			})
+		}
+	}
+	hasPGP := len(user.GetActivePGPKeys(true)) > 0
+	if !hasPGP {
+		suggestions = append(suggestions, ProofSuggestion{
+			Key:           "pgp",
+			ProfileText:   "Add a PGP key",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "PGP key",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		})
+	}
+	// Always show the option to create a new web proof.
+	suggestions = append(suggestions, ProofSuggestion{
+		Key:           "web",
+		ProfileText:   "Prove your website",
+		ProfileIcon:   dummyIcon,
+		PickerText:    "Your own website",
+		PickerSubtext: "",
+		PickerIcon:    dummyIcon,
+	})
+	if !user.IDTable().HasActiveCryptocurrencyFamily(libkb.CryptocurrencyFamilyBitcoin) {
+		suggestions = append(suggestions, ProofSuggestion{
+			Key:           "bitcoin",
+			ProfileText:   "Set a Bitcoin address",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Bitcoin address",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		})
+	}
+	if !user.IDTable().HasActiveCryptocurrencyFamily(libkb.CryptocurrencyFamilyZCash) {
+		suggestions = append(suggestions, ProofSuggestion{
+			Key:           "zcash",
+			ProfileText:   "Set a Zcash address",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Zcash address",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		})
+	}
+
+	// Alphabetize so that ties later on in SliceStable are deterministic.
+	sort.Slice(suggestions, func(i, j int) bool {
+		return suggestions[i].Key < suggestions[j].Key
+	})
+
+	// Priorities from the server.
+	serverPriority := make(map[string]int) // key -> server priority
+	maxServerPriority := 0
+	for _, displayConfig := range mctx.G().GetProofServices().ListDisplayConfigs() {
+		if displayConfig.Priority <= 0 {
+			continue
+		}
+		var altKey string
+		switch displayConfig.Key {
+		case "zcash.t", "zcash.z", "zcash.s":
+			altKey = "zcash"
+		case "http", "https", "dns":
+			altKey = "web"
+		}
+		serverPriority[displayConfig.Key] = displayConfig.Priority
+		if len(altKey) > 0 {
+			if _, ok := serverPriority[altKey]; !ok {
+				serverPriority[altKey] = displayConfig.Priority
+			}
+		}
+		if displayConfig.Priority > maxServerPriority {
+			maxServerPriority = displayConfig.Priority
+		}
+	}
+
+	// Fallback priorities for rows the server missed.
+	// Fallback priorities are placed after server priorities.
+	offlineOrder := []string{
+		"twitter",
+		"github",
+		"reddit",
+		"hackernews",
+		"rooter",
+		"web",
+		"pgp",
+		"bitcoin",
+		"zcash",
+		"~other~",
+	}
+	offlineOrderMap := make(map[string]int) // key -> offline priority
+	for i, k := range offlineOrder {
+		offlineOrderMap[k] = i
+	}
+
+	priorityFn := func(key string) int {
+		if p, ok := serverPriority[key]; ok {
+			return p
+		} else if p, ok := offlineOrderMap[key]; ok {
+			return p + maxServerPriority + 1
+		} else {
+			return offlineOrderMap["~other~"] + maxServerPriority + 1
```

Comment:
why not `len(offlineOrderMap) + maxServerPriority + 1` and ditch `~other~`

Reply:
this way you can move other, how cool is that. ditching